pkg/kv/kvserver/rangefeed/catchup_scan.go

```diff
@@ -145,24 +145,36 @@ func (i *CatchUpIterator) CatchUpScan(outputFn outputEventFn, withDiff bool) err
 
 		// Emit any new MVCC range tombstones when their start key is encountered.
 		// Range keys can currently only be MVCC range tombstones.
+		// We need to verify that the range tombstone is visible at the catch-up
+		// timestamp, since we might have come here after a call to NextIgnoringTime.
 		//
 		// TODO(erikgrinaker): Find a faster/better way to detect range key changes
 		// that doesn't involve constant comparisons. Pebble probably already knows,
 		// we just need a way to ask it.
 		if hasRange {
 			if rangeBounds := i.RangeBounds(); !rangeBounds.Key.Equal(rangeKeysStart) {
-				rangeKeysStart = append(rangeKeysStart[:0], rangeBounds.Key...)
+				// Clear scratch buffer, we'll populate it only when we need to. (This optimizes
+				// for the case in which there are few range tombstones visible)
+				rangeKeysStart = rangeKeysStart[:0]
```

Comment:
Isn't this going to cause it to trigger for every single point key that overlaps a range key?

Reply:
Yikes, good catch. Fixed.